### ISSUE TITLE
Office API CustomFunctions.StreamingInvocation Crashes when invocation.setResult() is trigger in the same cell as RTD function

### ISSUE THREAD

**Issue by jbqmGit** [OfficeDev/office-js#4083]
# Provide required information needed to triage your issue
Creating a custom function with a streaming invocation and invoking it to return a result in the same cell as an RTD function, causes Excel to crash. 
This occurs when we set the result on the StreamingInvocation, when invoking RTD in the same cell.

For example our custom formula signature is called QMS
Upon calling
`=QMS("goog", "last") + RTD("test", "test", "TEST")`
within a cell, we experience a crash

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC, Mac, Web
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: Office 2021 Version 2108, Office 365 Version 2401
* Operating System: Windows 10, Mac OS
* Browser (if using Office on the web): Chrome, Firefox, Edge
 
## Expected behavior

Upon writing
`=QMS("goog", "last") + RTD("test", "test", "TEST")`
Formula should be marked as invalid by excel

## Current behavior

Writing
`=QMS("goog", "last") + RTD("test", "test", "TEST")`
in a cell will cause excel to pause for 5 seconds then close and restart

## Steps to reproduce

1. create a custom formula using a streaming invocation 
and set the result
```
/**
 * This function returns real-time market data from a symbol based on the requested datapoint. Users must be subscribed into Quotesteam Active Sheets services
 * @CustomFunction
 * @description 
 * @helpUrl
 * @param {string} symbol 
 * @param {string} field 
 * @param {CustomFunctions.StreamingInvocation<string>} invocation custom function invocation, a necessary attribute for periodical update of the cell
 */
async function QMS(symbol, field, invocation) {
    invocation.setResult("ANY STRING");
}
```

2. Insert the custom formula into a cell with the rtd formula
 `=QMS("goog", "last") + RTD("test", "test", "TEST")`
4. Upon pressing enter, both formulas will invoke, and then cause a crash

## Link to live example(s)

# Provide additional details

## Context

We're currently migrating users from using Excel's RTD formula connected by a desktop client, to an independent add-in
Upon testing this interaction by chance, we found out that the RTD formula does not respond well when used with arithmetic alongside a streamingInvocation function. This is quite a large issue as our add-in is causing an excel crash, but there isn't much we can do on our side to prevent the issue, as it happens with the tools provided to use by office.js

## Useful logs

Logs are not provided upon crashing
![2024-02-01_467-467-467](https://github.com/OfficeDev/office-js/assets/158520501/3f5b431a-1a41-423c-93e8-28f15edec60c)
Here's a gif of how the application reacts.
- [ ] Console errors
- [x] Screenshots
- [ ] Test file (if only happens on a particular file)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by MiaofeiWang** [comment]
Hi @jbqmGit , thanks for reporting this. We are able to reproduce the issue and have created internal work item 8716200 to track it.

**Comment by MiaofeiWang** [comment]
Hi @jbqmGit , Sorry for the late response. There is a straight-forward workaround to avoid crash: JS streaming function in one cell, RTD function in another, then make any calculation with those 2 results. We hope to double confirm that your add-in users know this workaround to avoid crash.  
On Microsoft side, we are trying to fix the crash issue. But for the functionality of streaming JS custom function and RTD function in same cell, after some investigation, we cannot guarantee to make it work. Most probably such case will get a failure with "#VALUE" or some other error. Want to hear your feedback about this proposal. Thank you.

**Comment by govert** [comment]
Many .xll-based Excel add-ins use the RTD mechanism _inside_ the UDF implementation to provide async, streaming or 'handle'-based functions. I would presume the same bug between JS functions and the .xll UDFs with RTD might then occur. But in such cases the user can't easily see that a UDF they are calling is an 'RTD' UDF and can't currently be safely mixed with a JS UDF in one formula.

So, I think the suggested workaround is useful to know about, but not always easy to use as guidance.

**Comment by MiaofeiWang** [comment]
@govert Thanks for the scenario introduction which helps understand more clearly. We will try to work out a robust solution.

**Comment by XuanZhouMSFT** [comment]
We have fixed the crash issue and it should be available in build 16.0.17830 and later. Please have a try.
If it still repro for you, we will re-active.
